src/routes/give-away/create.js

```diff
@@ -0,0 +1,51 @@
+import models from '../../models'
+
+export default {
+
+  method: 'post',
+  path: '/giveaways?',
+
+  requireAuthentication: true,
+
+  handler(request, response) {
+
+    // @TODO: This is getting claned up by the orphan job
+    return models.CodexRecordFile.findById('5b1ef57078d795dc2bae3560')
+      .lean()
+      .then((leanMainImage) => {
+        delete leanMainImage._id
+
+        return models.CodexRecordFile.create(leanMainImage)
+      })
+      .then((giveawayMainImage) => {
+        const giveawayMetadata = {
+          creatorAddress: '0x0', // placeholder because it's required
+          codexRecordTokenId: 'giveaway', // placeholder so it doesn't get cleaned up by the remove-orphans job
+
+          name: 'Codex Original Art',
+          description: 'Designed by Seb',
+          mainImage: giveawayMainImage,
+        }
+
+        const newCodexRecordMetadata = new models.CodexRecordMetadata(giveawayMetadata)
+
+        return newCodexRecordMetadata.save()
+          .then(() => {
+            return newCodexRecordMetadata
+              .populate('mainImage images files') // TODO: move this to a post-save hook (but check that they haven't been populated already)
```

Comment:
This line (and the subsequent`.execPopulate()`) should be unnecessary since you'll already have the `mainImage` populated (since you created it then attached it), and the metadata has no `images` or `files`.

Reply:
Thanks. I pretty much copied as much as I could since it was working over there already :)